### ISSUE TITLE
nody-greeter --help outputs "segmentation fault (core dumped)"

### ISSUE THREAD

**Issue by lgor360** [JezerM/nody-greeter#58]
I decided to check this package for errors. I ran node-greeter --help, but got this error message. I went to the logs and saw...

kernel: audit: type=1400 audit(1732117727.597:180): apparmor="AUDIT" operation="userns_create" class="namespace" info="Userns create - transitioning profile" profile="unconfined" pid=12214 comm="nody-greeter" requested="userns_create" target="unprivileged_userns"

why. does --help have to be started from root?.. AND WHY --help COMMAND USER NAMES!?

**Comment by JezerM** [comment]
Lol, that shouldn't happen at all. There is no need to run nody-greeter as root, so this probably mean that the build process must have failed, probably the electron compilation with node-gtk.

Could you provide your build logs? (`npm run build` and share the output).

**Comment by lgor360** [comment]
> Lol, that shouldn't happen at all. There is no need to run nody-greeter as root, so this probably mean that the build process must have failed, probably the electron compilation with node-gtk.
> 
> Could you provide your build logs? (`npm run build` and share the output).

okay
it just shows a message like "building models..... 0" for 5 minutes and then abruptly SUCCESS!
If all logs are needed, please advise whether to run only npm run rebuild or something else.

**Comment by JezerM** [comment]
A screenshot or the full log would be helpful.

**Comment by lgor360** [comment]
> A screenshot or the full log would be helpful.

just npm run build?

**Comment by JezerM** [comment]
Yes. You could also try to remove the `build/unpacked` and `build/nody-asar` directories before `npm run build`.

**Comment by lgor360** [comment]
> Yes. You could also try to remove the `build/unpacked` and `build/nody-asar` directories before `npm run build`.

[logs.txt](https://github.com/user-attachments/files/17834344/logs.txt)

here. please do not delete this issue :)

**Comment by JezerM** [comment]
Everything seems alright... Try to run `npx electron . --help` inside the nody-greeter repo. It should show something like this:

<img width="742" alt="Captura de pantalla 2024-11-20 a la(s) 11 24 40" src="https://github.com/user-attachments/assets/f26f5492-884e-4cae-81ec-b8e68507aff5">

**Comment by JezerM** [comment]
If it doesn't work, then try to rebuild with `npm run rebuild` to rebuild electron with node-gtk. Afterwards, run `npx electron . --help` to check if it works.

**Comment by lgor360** [comment]
> Everything seems alright... Try to run `npx electron . --help` inside the nody-greeter repo. It should show something like this:
> 
> <img width="742" alt="Captura de pantalla 2024-11-20 a la(s) 11 24 40" src="https://github.com/user-attachments/assets/f26f5492-884e-4cae-81ec-b8e68507aff5">
> 

wait... you mean... go to the folder with the cloned repository and run this command?... ok....

**Comment by JezerM** [comment]
Yes

**Comment by lgor360** [comment]
> npx electron . --help

/home/USERNAME/nody-greeter/node_modules/electron/dist/electron exited with signal SIGSEGV

I did some research online about SIGSEGV and it looks like it's the same segmentation fault... noooooooooooooooooooooo :[

**Comment by lgor360** [comment]
> Yes

/home/USERNAME/nody-greeter/node_modules/electron/dist/electron exited with signal SIGSEGV

logs:
kernel: audit: type=1400 audit(1732174095.111:201): apparmor="AUDIT" operation="userns_create" class="namespace" info="Userns create - transitioning profile" profile="unconfined" pid=4273 comm="electron" requested="userns_create" target="unprivileged_userns"

I did some research online about SIGSEGV and it looks like it's the same segmentation fault... noooooooooooooooooooooo :[

**Comment by JezerM** [comment]
Hi, sorry for the late response, I've been busy...

So, I just looked for this issue and it seems it's an issue with recent versions of `apparmor` in Ubuntu 24.04+, where unknown namespaces like `com.github.jezerm.nody-greeter` are prohibited to be executed.

Here's an open issue on Electron's GitHub: https://github.com/electron/electron/issues/41066

Currently, you can execute the following command to disable AppArmor unprivileged user namespaces restriction:

```sh
sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
```

From my side, I'll try to add an AppArmor profile for nody-greeter (when I have the time to do so). Meanwhile, this issue will remain open.

**Comment by lgor360** [comment]
> Hi, sorry for the late response, I've been busy...
> 
> So, I just looked for this issue and it seems it's an issue with recent versions of `apparmor` in Ubuntu 24.04+, where unknown namespaces like `com.github.jezerm.nody-greeter` are prohibited to be executed.
> 
> Here's an open issue on Electron's GitHub: https://github.com/electron/electron/issues/41066
> 
> Currently, you can execute the following command to disable AppArmor unprivileged user namespaces restriction:
> 
> ```sh
> sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
> ```
> 
> From my side, I'll try to add an AppArmor profile for nody-greeter (when I have the time to do so). Meanwhile, this issue will remain open.

hi again and thanks!
